README.md

```diff
@@ -65,6 +65,59 @@ the module wide setting).
 * `@JsonSerializable` is the annotation used to generate `JsonAdapter`'s. Should only be placed on Kotlin data classes.
 * `@KotshiConstructor` should be used when there are multiple constructors in the class. Place it on the primary constructor.
 * `@KotshiJsonAdapterFactory` makes Kotshi generate a JsonAdapter factory. Should be placed on an abstract class that implements `JsonAdapter.Factory`.
+* `@JsonDefaultValue` used for enabling default values (see [below](#default-values))
+* `@JsonDefaultValueString` used for specifying default values for String properties inline
+* `@JsonDefaultValueBoolean` used for specifying default values for Boolean properties inline
+* `@JsonDefaultValueByte` used for specifying default values for Byte properties inline
+* `@JsonDefaultValueChar` used for specifying default values for Char properties inline
+* `@JsonDefaultValueShort` used for specifying default values for Short properties inline
+* `@JsonDefaultValueInt` used for specifying default values for Int properties inline
+* `@JsonDefaultValueLong` used for specifying default values for Long properties inline
+* `@JsonDefaultValueFloat` used for specifying default values for Float properties inline
+* `@JsonDefaultValueDouble` used for specifying default values for Double properties inline
+
+### Default values
+You can use default values by first annotating a function, field, constructor or enum type with the annotation
+`@JsonDefaultValue`. This will be the provider of the default value.
+
+You then annotate a parameter of the same type (or a supertype) with the same annotation.
+
+If you need to have multiple default values of the same type you can create a custom default value annotation by
+annotating it with `@JsonDefaultValue`.
+
+If you don't want to define default value providers for primitivie and string properties you can use the specialized 
```

Comment:
`primitive`

Reply:
Also, maybe worth moving this below the `@JsonDefaultValue` code block and adding an example code block that shows off the simpler approach

Reply:
The typo was fixed. I mentioned the annotations in the list above and I already included one example in the example block.

Reply:
Ah, I missed that you could mix&match the approaches, cool.